RELEASE.md

```diff
@@ -1,33 +1,68 @@
-## Making a release of the Checker Framework Gradle plugin
+## Making and merging pull requests
 
-This document describes how to make a release of the Checker Framework
-Gradle plugin, which this repository contains.
+The Checker Framework Gradle plugin (which this repository contains)
+uses a continuous-delivery style of releases: any time that the
+behavior of the plugin changes, a new release is cut.
 
-### Prequisites
+### Making a pull request
+
+All pull requests should be made from a feature branch. Never push
+directly to `master`.
+
+If your pull request will change any behavior
+of the plugin (including bug fixes and new features), you should
+update the version string. When changing only documentation
+(e.g. this file or the README), do not update the version string.
+
+After you have made the changes you wish to merge into `master`,
+you should:
+1. Choose a new version string. Please try to respect
+[semantic versioning](https://semver.org/). Do not augment the major
+version without explicit approval from all the maintainers.
+Try to avoid version numbers that will conflict with 
+[other active pull requests](https://github.com/kelloggm/checkerframework-gradle-plugin/pulls).
+2. Search for the current plugin version
+in the directory containing this file, and replace all instances of it
+with the new version string. Commit the result.
+3. Push to your feature branch (if you have not yet created one, do so. All development
+should happen on feature branches - never on `master`).
+4. Make a pull request against the `master` branch.
+
+### Merging a pull request
+
+#### Prequisites
 
 * Copy the publishing credentials to your `~/.gradle/gradle.properties` file.
 The credentials are stored in the same place as the credentials used to make
 a Checker Framework release, in a `gradle.properties` file.
 To gain access to the credentials, contact one of the maintainers privately.
 
-### Release process
+#### Release process
 
-1. Ensure that the
+1. Review the pull request. If you approve it, proceed.
+2. Ensure that the
 [Travis build](https://travis-ci.com/kelloggm/checkerframework-gradle-plugin/branches)
-is passing on the `master` branch.
-2. Ensure that you have checked out the `master` branch of the project. Do
-not make releases from other branches.
-3. Update the plugin to use the [latest version](https://github.com/typetools/checker-framework/blob/master/changelog.txt) of the Checker Framework:
+passes in the pull request.
+3. Ensure that the new version string in the pull request is reasonable - it may
+have conflicted with another pull request that has already been merged. Check 
```

Comment:
If there is a conflict, that will be reflected in the pull request (there will be a conflict).  The problem is if there is no conflict.

Reply:
That is why the instructions tell the merger to check the released versions. I will clarify the wording, however.